File: app/src/main/java/com/abishov/hexocat/commons/schedulers/SchedulerProvider.java
package com.abishov.hexocat.commons.schedulers;

import io.reactivex.Scheduler;

/**
 * Used to decouple schedulers from RxJava chains in order to make
 * them unit-testable.
 */
public interface SchedulerProvider {
    Scheduler computation();

    Scheduler io();

    Scheduler ui();
}

File: app/src/main/java/com/abishov/hexocat/common/schedulers/SchedulerProvider.kt
package com.abishov.hexocat.common.schedulers

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

interface SchedulerProvider {

  fun computation(): Scheduler = Schedulers.computation()

  fun io(): Scheduler = Schedulers.io()

  fun ui(): Scheduler = AndroidSchedulers.mainThread()
}
